src/db.c

```diff
@@ -259,11 +260,19 @@ dict *getDict(redisDb *db, sds key) {
  * ownership of the SDS string, otherwise 0 is returned, and is up to the
  * caller to free the SDS string. */
 int dbAddRDBLoad(redisDb *db, sds key, robj *val) {
-    dict *d = getDict(db, key);
+    int slot = getKeySlot(key);
+    dict *d = db->dict[slot];
+    int was_empty = dictIsEmpty(d);
     dictEntry *de = dictAddRaw(d, key, NULL);
     if (de == NULL) return 0;
     dictSetVal(d, de, val);
     db->key_count++;
+    /* If dict transitioned from empty to non-empty, we should add it to the list of owned slots. */
+    if (was_empty) {
+        uint8_t success = 0;
+        db->owned_slots = intsetAdd(db->owned_slots, slot, &success);
```

Comment:
i think we should not allocate and populate that data structure (intset) in case we only have one dict (non-cluster mode)

Reply:
Ok, I'll change it. The reason why I did it this way, was to avoid custom logic (for cluster mode disabled) in the iterator, but it's quite easy to do it either way.